apps/dashboard/src/app/(main)/(protected)/projects/[projectId]/teams/[teamId]/page-client.tsx

```diff
@@ -1,8 +1,46 @@
 "use client";
-import { useAdminApp } from '../../use-admin-app';
+import { TeamMemberSearchTable } from '@/components/data-table/team-member-search-table';
+import { TeamMemberTable } from '@/components/data-table/team-member-table';
+import { ServerTeam } from '@stackframe/stack';
+import { ActionDialog, Button, useToast } from '@stackframe/stack-ui';
 import { notFound } from 'next/navigation';
 import { PageLayout } from '../../page-layout';
-import { TeamMemberTable } from '@/components/data-table/team-member-table';
+import { useAdminApp } from '../../use-admin-app';
+import { KnownErrors } from '@stackframe/stack-shared';
+
+
+export function AddUserDialog(props: {
+  open?: boolean,
+  onOpenChange?: (open: boolean) => void,
+  trigger?: React.ReactNode,
+  team: ServerTeam,
+}) {
+  const { toast } = useToast();
+
+  return <ActionDialog
+    title="Add a user"
+    trigger={props.trigger}
+  >
+    <TeamMemberSearchTable
+      action={(user) => <div className='flex justify-end w-[60px]'><Button
+        variant="outline"
+        onClick={async () => {
+          try {
+            await props.team.addUser(user.id);
+            toast({ title: 'User added to team', variant: 'success' });
+          } catch (error) {
+            if (error instanceof KnownErrors.TeamMembershipAlreadyExists) {
+              toast({ title: 'User already a member of this team', variant: 'destructive' });
```

Comment:
Actually, I think it is better to show an "added" button (see the updated image in the PR description). It will be a bit more annoying when we later paginate the team-member tables, but we can worry about that later

Reply:
What happens if the user was not added successfully because they're already in the team?

Reply:
SGTM

Reply:
The toast is not necessary though — from a user perspective, "the user has been added" and "the user has already been on the team" is equivalent